lib/trino-record-decoder/src/main/java/io/trino/decoder/json/DefaultJsonFieldDecoder.java

```diff
@@ -46,6 +46,13 @@
     private final long minValue;
     private final long maxValue;
 
+    protected DefaultJsonFieldDecoder()
```

Comment:
it's a bit tricky, I'm trying to extend DefaultJsonFieldDecoder to reduce code duplication, while I do want to utilize the current constructor, but it can only invoke  `isSupportedType` in DefaultJsonFieldDecoder, which will prevent us from supporting new types, I'm adding this default constructor as workaround so we can skip `isSupportedType` when calling `super()`, any suggestion?

Reply:
> any suggestion?

Yes, please undo ;)

Please make `PulsarJsonFieldDecoder` to implement `JsonFieldDecoder` without using `DefaultJsonFieldDecoder`.